### PR TITLE
Fix: Refresh status bar after resume

### LINKED ISSUE #2091: Info line not updated after standy
- KOReader version: koreader-kobo-arm-linux-gnueabihf-v2015.11-431-gc0846ee.zip
- Device: Kobo
#### Issue

Neither the battery status nor time are updated when switching on after standby. I have to turn the page first.

### PR DESCRIPTION
**Pull request by robert00s**
Close #2091 
This commit should fix problem with refresh info line (status bar) after resume KO.

### CODE REVIEW

**Comment by poire-z** [comment]
It's a totally logical fix.
But I'm used and like to see, when I resume, the time it was when I last read and suspended it. Updating this old time to the new time was just a matter of a diagonal swipe. I'll miss that :)

**Comment by robert00s** [comment]
I can add in settings -> status bar new options "Disable refresh footer after resume".
Defaults it will be disabled.

**Comment by poire-z** [comment]
No, I dont think it needs an option, your fix is the right thing. Unless other people than me join and express the same habbit.
Or it could also be done with a small delay, with `UIManager:scheduleIn(1, self.view.footer:updateFooter)`, enough to see the suspend time transitionning to the now time (but some people may consider this delay a bug).

**Comment by Frenzie** [comment]
@Hzj-jie On your second point, are you saying that updateBatteryState() isn't automatically triggered by updateFooter()?

**Comment by AlanSP1** [comment]
> I can add in settings -> status bar new options "Disable refresh footer after resume".
Defaults it will be disabled.
About option to control refresh status, it may be better to construct "positive" options, i.e. construct text to sound "Enable refresh footer after resume" and make it enabled by default. I think this is more logical and easier to comprehend from psychological perspective.

And, is it good to have footer refreshed? Well, it really depends on peoples habits. But, I think that it may be good to have option not to refresh it, or option to have short timer (something like 10 seconds or such).

In documentation I hope we will have there could be explained that this isn't bug, but feature.

**Comment by Frenzie** [comment]
What value does the outdated time/battery have?

**Comment by AlanSP1** [comment]
Sorry, didn't understand question?

But, it may be useful, I see that. For time, to know when we read last time and it also can help track battery dissipation while reader is sleeping.

**Comment by Frenzie** [comment]
Hm, seems like that should just be put in a log or something instead? I
don't really mind having to turn back a page and forward again to get the
updated info (usually I don't care) but imho the lack of updates is not a
feature.

tl;dr I support this PR.

**Comment by Hzj-jie** [comment]
@Frenzie, updateBatteryState does,
self.ui.document:setBatteryState(state)
But updateFooter() does something (more complex) else.

**Comment by Hzj-jie** [comment]
I support the decision of this change.

To;Dr:
For battery stats, a PR has just been merged: a battery stats plugin can track it.
For the last read time, I would also suggest to use Statistics plugin to achieve the goal. Though we do not have this feature in it.
Personally I have no strong negative impression if the display is not refresh immediately after refresh  , as I can understand refresh of e-ink is slow and inefficient. If we want to do it, a very short delay can be understand (in code, UIManager:nextTick()), I.e unblock other regular work. But 10 seconds seems like a too long time. But a true-by-default opinion is acceptable, though I prefer to have an advanced configuration. I believe this setting is not likely to be changed by regular users.

File: frontend/apps/reader/modules/readerrolling.lua [hunk 1]
```diff
@@ -332,7 +332,7 @@ function ReaderRolling:onPageUpdate(new_page)
 end
 
 function ReaderRolling:onResume()
-    self:updateBatteryState()
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerrolling.lua]: Okay, so don't remove this line 'cause it's not redundant according to @Hzj-jie ;-)

**Comment by robert00s** [comment]
This code:

```
function ReaderFooter:onResume()
    UIManager:scheduleIn(1, self:updateFooter())
end
```
generates from time to time on my Kindle PW3 that error (crash):
```
./luajit: frontend/ui/uimanager.lua:455: attempt to call field 'action' (a nil value)
stack traceback:
	frontend/ui/uimanager.lua:455: in function '_checkTasks'
	frontend/ui/uimanager.lua:604: in function 'handleInput'
	frontend/ui/uimanager.lua:698: in function 'run'
	./reader.lua:188: in main chunk
	[C]: at 0x0000bef1
lipc-wait-event exited normally with status: 0 
```
so I stay:

```
function ReaderFooter:onResume()
    self:updateFooter()
end
```

**Comment by poire-z** [comment]
> generates from time to time on my Kindle PW3 that error (crash)

Try it without the () to updateFooter , as:
```lua
function ReaderFooter:onResume()
    UIManager:scheduleIn(1, self:updateFooter)
end
```
(scheduleIn expects the 2nd argument to be something `callable` , a function, the function itself. When you use self:updateFooter(), u're actually calling the function and giving the result of this function as the 2nd argument to schedulIn).
I think this suggestion may not even work, because of the ":", so you may try to wrap it (with the () this time) in a real simple function:
```lua
function ReaderFooter:onResume()
    UIManager:scheduleIn(1, function() self:updateFooter() end)
end
```

**Comment by houqp** [comment]
If you need to access self, you need to wrap it in a anonymous function call like the example @poire-z gave.

Battery stats is already tracked by @Hzj-jie 's new battery stats plugin, so it's better to use that plugin's log to get insights into battery usage.
